core/src/main/scala/org/apache/spark/serializer/KryoSerializer.scala

```diff
@@ -92,6 +94,16 @@ class KryoSerializer(conf: SparkConf)
       new KryoOutput(bufferSize, math.max(bufferSize, maxBufferSize))
     }
 
+  @transient
+  private lazy val factory: KryoFactory = new KryoFactory() {
```

Comment:
I think you're welcome to just write ...
```
private lazy val factory = new KryoFactory() {
  override def create: Kryo = newKryo()
}
```
but it doesn't matter.

Reply:
Thanks, I'll remove, I had a problem previously but it looks like the @transient on the pool is enough itself.